### DISCUSSION TITLE
How to know if read only property can be set in constructor?

### DISCUSSION THREAD

**Discussion by gao-artur** [dotnet/roslyn#71743] [General] [answered]
Given the `IPropertySymbol` for the following properties:
```c#
// can be set in the constructor
public string P1 { get; }
// can't be set in constructor
public string P2 => "zzz";
```

In both cases, the `SetMethod is null` and `IsReadOnly is true`. How can I find out which one can be set in the constructor?

**Comment by CyrusNajmabadi** [comment, marked as answer]
Look at the property and see if there is an associated field.  If so, it is an autoprop and can be set

**Reply by gao-artur** [reply]
This helped. Thanks!
I found [this](https://github.com/dotnet/roslyn/blob/2fe96bca1092f880e91eea6eb17ea3487d89309a/src/Features/Core/Portable/EditAndContinue/Utilities/Extensions.cs#L186-L187) implementation:
```c#
public static bool IsAutoProperty(this IPropertySymbol property)
    => property.ContainingType.GetMembers().Any(static (member, property) => member is IFieldSymbol field && field.AssociatedSymbol == property, property);
```
